tests/Agency/StorePerformanceTest.cpp

```diff
@@ -0,0 +1,651 @@
+    try {
+      auto expected{VPackParser::fromJson(expected_result)};
+      if (!velocypack::NormalizedCompare::equals(result->slice(), expected->slice())) {
+        throw std::runtime_error(
+            result->toJson() + " should have been equal to " + expected->toJson());
+      }
+    } catch (std::exception& ex) {
+      throw std::runtime_error(std::string(ex.what()) + " comparing to " + expected_result);
+    }
+  }
+
+  template <typename TSource>
+  static std::string to_json_object(
+      TSource const& src,
+      std::function<std::string(typename TSource::const_reference)> extractName,
+      std::function<std::string(typename TSource::const_reference)> extractValue) {
+    bool is_first{true};
+    return "{" +
+           std::accumulate(src.begin(), src.end(), std::string(),
+                           [&is_first, extractName,
+                            extractValue](std::string partial, auto const& element) {
+                             if (is_first) {
+                               is_first = false;
+                             } else {
+                               partial += ",";
+                             }
+                             partial += "\"" + extractName(element) +
+                                        "\": " + extractValue(element);
+                             return partial;
+                           }) +
+           "}";
+  }
+
+  template <typename TSource>
+  static std::string to_json_object(TSource const& src) {
+    return to_json_object(
+        src, [](auto const& element) { return element.first; },
+        [](auto const& element) { return element.second; });
+  }
+
+  void threaded_writes(std::function<std::string()> value_generator) {
+    auto const count {repetition_times[0]};
+    std::vector<std::thread> threads;
+    OperationMeasurement::Observations points{1 + count};
+    std::mutex m_ready, m_start;
+    std::condition_variable cv_ready, cv_start;
+    std::atomic_uint worker_count{0};
+    std::generate_n(std::back_inserter(threads), count, [this, &m_start, &cv_start, &cv_ready, &worker_count, &points, value_generator](){ 
+      return std::thread([this, &m_start, &cv_start, &cv_ready, &worker_count, &points, value_generator]() {
+        uint const thread_index {worker_count++};
+        auto const write_query {VPackParser::fromJson(value_generator())};
+        std::unique_lock<std::mutex> lk{m_start};
+        cv_ready.notify_one();
+        cv_start.wait(lk);
+        for (size_t i{}; i < repetition_times[1]; ++i) {
+          write(write_query);
+        }
+        points.at(1 + thread_index) = OperationMeasurement::Clock::now();
+      });});
+    std::unique_lock<std::mutex> lk{m_ready};
+    cv_ready.wait(lk, [&worker_count, count](){ 
+      return worker_count.load() == count;
+    });
+    
+    std::this_thread::sleep_for(std::chrono::milliseconds{50});
+    points[0] = OperationMeasurement::Clock::now();
+    cv_start.notify_all();
```

Comment:
This startup and finish is nice. It takes the thread creation latency out of the equation, which is exactly, what we want.

Reply:
Thanks.